engine/execution/state/state.go

```diff
@@ -283,24 +283,48 @@ func (s *state) SaveExecutionResults(
 		trace.EXEStateSaveExecutionResults)
 	defer span.End()
 
+	header := result.ExecutableBlock.Block.Header
+
+	//outside batch because it requires read access
+	err := s.UpdateHighestExecutedBlockIfHigher(childCtx, header)
+	if err != nil {
+		return fmt.Errorf("cannot update highest executed block: %w", err)
+	}
+	return nil
+}
+
+func (s *state) saveExecutionResults(
+	ctx context.Context,
+	result *execution.ComputationResult,
+) error {
```

Comment:
nit: Maybe we can use a named return value for error here, to make error checking in `defer` less fragile.

```suggestion
) (err error) {
```